app/Http/Controllers/admin/BlogController.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace App\Http\Controllers\admin;
+
+use App\Http\Controllers\Controller;
+use Illuminate\Http\Request;
+use App\Models\Category;
+use App\Models\User;
+use App\Models\Blog;
+use App\Http\Requests\blog\StoreRequest;
+use App\Http\Requests\blog\UpdateRequest;
+
+class BlogController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $data = Blog::orderBy('created_at', 'DESC')->paginate(5);
+        if($search = request()->search){
+            $data = Blog::orderBy('created_at', 'DESC')->where('title','like','%'.$search.'%')->paginate(5);
+        }
```

Comment:
```suggestion
        $data = Blog::orderBy('created_at', 'DESC')->paginate(5);
        if($search = request()->search){
            $data = $data->where('title','like','%'.$search.'%');
        }
```
Thử kiểu này xem dc k em

Reply:
dạ vâng cách này cho ra cách làm ổn định với paginate hơn cái hôm trước em làm ạ

Reply:
cách này sẽ gây ra lỗi khi tìm kiếm ạ

Reply:
nó báo lỗi vì hàm paginate không được sử dụng khi đổ kết quả tìm kiếm ra ạ

Reply:
$search = request()->search;
$data = Blog::orderBy('created_at', 'DESC')->where('title','like','%'.$search.'%')->paginate(5);
return view('admin.blogs.index',compact('data','search'));
em có thử cách này để thay cả đoạn kia nhưng kết quả cho ra là khi đổ dữ liệu ra nó bị không đầy đủ theo thứ tự bảng ạ